lib/_http_outgoing.js

```diff
@@ -337,6 +337,14 @@ OutgoingMessage.prototype.getHeader = function(name) {
 };
 
 
+OutgoingMessage.prototype.getAllHeaders = function() {
+  if (!this._headers)
+    return;
```

Comment:
Does this mean that I always should check a type of the returning value? Are there cases when you need to check that at least one header was set? 
I think that it would be better to have separate method to check this. Outgoing message can be immediately initialized with `_headers = {}` and it will allow to do not check every `setHeader()` call that the property was initialized